server/models/User.py

```diff
@@ -0,0 +1,24 @@
+from datetime import datetime
+from server import create_db, login_manager
+from flask_login import UserMixin
+
+db = create_db()
+
+login_manager.login_view = "login_method"
+login_manager.login_message_category = "info"
+
+
+@login_manager.user_loader
+def load_user(user_id):
+    return User.query.get(int(user_id))
+
+
+class User(db.Model, UserMixin):
+    id = db.Column(db.Integer, primary_key=True)
+    username = db.Column(db.String(20), unique=True, nullable=False)
+    email = db.Column(db.String(120), unique=True, nullable=False)
+    image_file = db.Column(db.String(20), nullable=False, default="default.png")
+    password = db.Column(db.String(60), nullable=False)
```

Comment:
I think we can let these three as `Text`, see:
- For e-mails, although the latest RFC about it states the length limit is 254, we can let it as unlimited (since who knows when a new RFC will extend the limit);
- For image files, I'm not sure we have a reason to limit the length. Also, calling `image_url` might fit better;
- For passwords, 60 is only valid for `bcrypt` with the latest algorithm (2b) and with steps >=10 and <= 99. If we use >=100 or any upcoming algorithm (say, 3), 60 might not be exactly what we expect. `Text` would avoid this issue as well.

```suggestion
    email = db.Column(db.Text, unique=True, nullable=False)
    image_url = db.Column(db.Text, nullable=False, default="default.png")
    password = db.Column(db.Text, nullable=False)
```

Reply:
Oh, btw, the "image" would be the avatar?

Reply:
> I think we can let these three as `Text`, see:
> 
> * For e-mails, although the latest RFC about it states the length limit is 254, we can let it as unlimited (since who knows when a new RFC will extend the limit);
> * For image files, I'm not sure we have a reason to limit the length. Also, calling `image_url` might fit better;
> * For passwords, 60 is only valid for `bcrypt` with the latest algorithm (2b) and with steps >=10 and <= 99. If we use >=100 or any upcoming algorithm (say, 3), 60 might not be exactly what we expect. `Text` would avoid this issue as well.

Completely agree with your suggestions. I'll do the necessary changes!

Reply:
> Oh, btw, the "image" would be the avatar?

Yeah. Any User will be having a default avatar until they change it.
If we do not plan to keep Profile Picture of User, I'll remove it but having one would be great

Reply:
> > Oh, btw, the "image" would be the avatar?
> 
> Yeah. Any User will be having a default avatar until they change it.
> If we do not plan to keep Profile Picture of User, I'll remove it but having one would be great

Oh, you can keep it :) Was just trying to understand it properly.